app/services/identifier_service.rb

```diff
@@ -47,6 +49,22 @@ class RestrictedArkError < StandardError; end
     "http://findingaids.princeton.edu/collections/#{resource.source_metadata_identifier.first.tr('_', '/')}"
   end
 
+  private_class_method def self.geo_metadata(resource)
+    slug = "princeton-#{resource.identifier.first.gsub(%r(ark:/\d{5}/), '')}"
+    metadata(resource).merge(target: "https://maps.princeton.edu/catalog/#{slug}")
```

Comment:
Is this just customizing the URL for the Geo resources?  Couldn't this be done in the `url_for` method? 
Or am I missing something here?

Reply:
Generating the slug for the Pulmap URL requires the ARK identifier to be minted first. The code here mints the ARK using the default target URL and then updates the URL to the correct value afterwards.

Reply:
👍 OK, that makes sense.